src/cargo/util/errors.rs

```diff
@@ -1,101 +1,148 @@
+//     }
+//
+//     links {
+//         CrateRegistry(registry::Error, registry::ErrorKind);
+//     }
+//
+//     foreign_links {
+//         ParseSemver(semver::ReqParseError);
+//         Semver(semver::SemVerError);
+//         Ignore(ignore::Error);
+//         Io(io::Error);
+//         SerdeJson(serde_json::Error);
+//         TomlSer(toml::ser::Error);
+//         TomlDe(toml::de::Error);
+//         ParseInt(num::ParseIntError);
+//         ParseBool(str::ParseBoolError);
+//         Parse(string::ParseError);
+//         Git(git2::Error);
+//         Curl(curl::Error);
+//     }
+//
+//     errors {
+//         Internal(err: Box<CargoErrorKind>) {
+//             description(err.description())
+//             display("{}", *err)
+//         }
+//         ProcessErrorKind(proc_err: ProcessError) {
+//             description(&proc_err.desc)
+//             display("{}", &proc_err.desc)
+//         }
+//         CargoTestErrorKind(test_err: CargoTestError) {
+//             description(&test_err.desc)
+//             display("{}", &test_err.desc)
+//         }
+//         HttpNot200(code: u32, url: String) {
+//             description("failed to get a 200 response")
+//             display("failed to get 200 response from `{}`, got {}", url, code)
+//         }
+//     }
+// }
+
+// impl CargoError {
+//     pub fn into_internal(self) -> Self {
+//         CargoError(CargoErrorKind::Internal(Box::new(self.0)), self.1)
+//     }
+//
+//     fn is_human(&self) -> bool {
+//         match self.0 {
+//             CargoErrorKind::Msg(_) |
+//             CargoErrorKind::TomlSer(_) |
+//             CargoErrorKind::TomlDe(_) |
+//             CargoErrorKind::Curl(_) |
+//             CargoErrorKind::HttpNot200(..) |
+//             CargoErrorKind::ProcessErrorKind(_) |
+//             CargoErrorKind::CrateRegistry(_) => true,
+//             CargoErrorKind::ParseSemver(_) |
+//             CargoErrorKind::Semver(_) |
+//             CargoErrorKind::Ignore(_) |
+//             CargoErrorKind::Io(_) |
+//             CargoErrorKind::SerdeJson(_) |
+//             CargoErrorKind::ParseInt(_) |
+//             CargoErrorKind::ParseBool(_) |
+//             CargoErrorKind::Parse(_) |
+//             CargoErrorKind::Git(_) |
+//             CargoErrorKind::Internal(_) |
+//             CargoErrorKind::CargoTestErrorKind(_) |
+//             CargoErrorKind::__Nonexhaustive { .. } => false
+//         }
+//     }
+// }
+
+#[derive(Debug, Fail)]
+#[fail(display = "failed to get 200 response from `{}`, got {}", url, code)]
+pub struct HttpNot200 {
+    pub code: u32,
+    pub url: String,
+}
+
+pub struct Internal {
+    inner: Error,
+}
+
+impl Internal {
+    pub fn new(inner: Error) -> Internal {
+        Internal { inner }
     }
 }
 
-impl CargoError {
-    pub fn into_internal(self) -> Self {
-        CargoError(CargoErrorKind::Internal(Box::new(self.0)), self.1)
+impl Fail for Internal {
```

Comment:
Will keep in mind that this can't be derived and see if I could make it so it could be. :-)